server/channels/app/report.go

```diff
@@ -64,22 +64,24 @@ func (a *App) compileCSVChunks(prefix string, numberOfChunks int, headers []stri
 	if err = w.Error(); err != nil {
 		return model.NewAppError("saveCSVChunk", "app.save_csv_chunk.write_error", nil, "", http.StatusInternalServerError).Wrap(err)
 	}
-	_, appErr := a.WriteFile(&headerBuf, filePath)
-	if appErr != nil {
-		return appErr
-	}
 
 	for i := 0; i < numberOfChunks; i++ {
 		chunkFilePath := makeFilePath(prefix, i, "csv")
 		chunk, err := a.ReadFile(chunkFilePath)
 		if err != nil {
 			return err
 		}
-		if _, err = a.AppendFile(bytes.NewReader(chunk), filePath); err != nil {
+		_, writeErr := compiledBuf.Write(chunk)
```

Comment:
`AppendFile` was written assuming that we were uploading large potentially multipart files, and is designed to work on bigger parts. Our S3 library enforces a minimum part size of 5MB, so since most of our reports for the time being are less than that, I think we're safe to remove this very slight optimization.